pkg/apis/v1alpha1/subnetport_types.go

```diff
@@ -0,0 +1,83 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetPortSpec defines the desired state of SubnetPort.
+type SubnetPortSpec struct {
+	// Subnet defines the parent Subnet name of the SubnetPort.
+	Subnet string `json:"subnet,omitempty"`
+	// SubnetSet defines the parent SubnetSet name of the SubnetPort.
+	SubnetSet string `json:"subnetset,omitempty"`
+	// Attachment refers to the virtual machine which the SubnetPort is attached.
+	Attachment corev1.TypedLocalObjectReference `json:"attachment,omitempty"`
+}
+
+// SubnetPortStatus defines the observed state of SubnetPort.
+type SubnetPortStatus struct {
+	// Conditions describes current state of SubnetPort.
+	Conditions []SubnetPortCondition `json:"conditions,omitempty"`
+	// InterfaceID describes the attachment VIF ID owned by the SubnetPort in NSX-T.
+	InterfaceID string `json:"interfaceID,omitempty"`
```

Comment:
What is the terminology in NSX-T API? I guess we should use either VIFID to be consistent with NSX-T API, or ExternalID to be consistent with VAPI. @lxiaopei

Reply:
In NSX API, it is the id in Port Attachment. Shall we name it PortAttachmentID or VIFID? Which do you prefer? @jianjuns 
- !Type
  id: PortAttachment
  title: Attachment information on the port
  type: object
  description: Detail information about port attachment
  properties:
    id:
      title: Port attachment ID
      type: string
      description: VIF UUID on NSX Manager. If the attachement type is PARENT, this
        property is required.
      required: false

Reply:
Probably we can change to VIFUUID. @heypnus what do you think?

Reply:
Renamed it to `VIFID`, thanks.